ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/customercenter/navigation/CustomerCenterNavigationState.kt

```diff
@@ -0,0 +1,68 @@
+package com.revenuecat.purchases.ui.revenuecatui.customercenter.navigation
+
+import androidx.compose.runtime.Immutable
+import java.util.ArrayDeque
+import java.util.Deque
+
+@Immutable
+internal data class CustomerCenterNavigationState(
+    val backStack: Deque<CustomerCenterDestination> = ArrayDeque<CustomerCenterDestination>().apply {
+        push(CustomerCenterDestination.Main)
+    },
+) {
+    val currentDestination: CustomerCenterDestination
+        get() = backStack.peek() ?: CustomerCenterDestination.Main
+
+    val canNavigateBack: Boolean
+        get() = backStack.size > 1
+
+    fun push(destination: CustomerCenterDestination): CustomerCenterNavigationState {
```

Comment:
Do we want to add some tests for all this logic? Might also be good to add UI tests for an actual navigation I guess if possible...

Reply:
Yes! Good point. Will add some tests